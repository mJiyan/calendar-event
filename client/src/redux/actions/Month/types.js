const SET_MONTH = 'MONTH/SET_MONTH';

const actionTypes = {
  SET_MONTH,
};

export default actionTypes;
