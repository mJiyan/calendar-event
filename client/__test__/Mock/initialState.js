import dayjs from 'dayjs';

const initialState = {
  EventReducer: {
    events: [
      {
        _id: '619f90f7abfdf7635dc91f43',
        day: 1636531800000,
        description: 'Meet with friends',
        id: '619f90f7abfdf7635dc91f43',
        label: 'purple',
        title: 'Meeting',
      },
      {
        _id: '61cd677c3254c8001e8f0769',
        day: dayjs(),
        description: 'Buy new notebook',
        id: '61cd677c3254c8001e8f0769',
        label: 'blue',
        title: 'Notebook',
      },
    ],
    event: {},
    selectedEvent: {
      _id: "61cd8200bc6749001f5bf489",
      title: "Meeting",
      description: "Meet with friends",
      label: "purple",
      day: 1638392400000,
      __v: 0,
      id: "61cd8200bc6749001f5bf489"
    },
  },
  MonthReducer: {
    month: 1
  },
  DayReducer: {
    day: dayjs()
  },
  ModalReducer: {
    modal: false
  }
};


export default initialState;