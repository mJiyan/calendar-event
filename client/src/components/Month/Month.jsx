import React from 'react';
import Day from '../Day/Day';

const Month = ({ month }) => (
  <div className="flex-1 grid grid-cols-7 grid-rows-5">
    {month.map((row, rowIndex) => (
      <React.Fragment key={rowIndex}>
        {row.map((day, dayIndex) => (
          <Day day={day} key={dayIndex} rowIndex={rowIndex} />
        ))}
      </React.Fragment>
    ))}
  </div>
);

export default Month;
