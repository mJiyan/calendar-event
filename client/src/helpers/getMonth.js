import dayjs from 'dayjs';

export function getMonth(month = dayjs().month()) {
  const year = dayjs().year();

  const firstDayInMonth = dayjs(new Date(year, month, 0)).day();

  let presentMonth = 0 - firstDayInMonth;
  /** 5 week 7 day */
  const days2DArray = new Array(5).fill([]).map(() =>
    new Array(7).fill(null).map(() => {
      presentMonth++;
      return dayjs(new Date(year, month, presentMonth));
    }),
  );

  return days2DArray;
}
