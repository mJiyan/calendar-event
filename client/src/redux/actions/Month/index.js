import actionTypes from './types';

export const setMonthIndex = (month) => async (dispatch) => {
  dispatch({
    type: actionTypes.SET_MONTH,
    payload: month,
  });
};
