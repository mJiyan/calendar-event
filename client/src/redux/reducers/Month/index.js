import dayjs from 'dayjs';
import actionTypes from '../../actions/Month/types';

const initialState = {
  month: dayjs().month(),
};

const MonthReducer = (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.SET_MONTH:
      return { ...state, month: action.payload };
    default:
      return state;
  }
};

export default MonthReducer;
