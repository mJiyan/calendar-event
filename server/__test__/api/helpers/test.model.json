{
    "event": {
        "title": "Interview",
        "description": "Interview with customer",
        "label": "red",
        "day": 1637654400000
    }
}