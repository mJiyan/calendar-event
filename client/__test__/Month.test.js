import React from 'react';
import Month from '../src/components/Month/Month';
import { mount, shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { getMonth } from '../src/helpers/getMonth';
import renderer from 'react-test-renderer';
import initialState from './Mock/initialState';

const mockStore = configureMockStore([thunk]);
const store = mockStore(initialState);
const currentMonth = getMonth(1);

describe('rendering components', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Month month={currentMonth} />
    </Provider>,
  );

  test('it should match with the month component', () => {
    expect(component).toMatchSnapshot();
  });

});
