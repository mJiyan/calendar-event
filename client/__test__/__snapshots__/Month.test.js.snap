// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering components it should match with the month component 1`] = `
<div
  className="flex-1 grid grid-cols-7 grid-rows-5"
>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-31-01-22"
    >
      <p
        className="text-sm p-1 mt-1"
        day-testid="row-value-31-01-22"
      >
        MON
      </p>
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-31-01-22"
      >
        31
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-31-01-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-01-02-22"
    >
      <p
        className="text-sm p-1 mt-1"
        day-testid="row-value-01-02-22"
      >
        TUE
      </p>
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-01-02-22"
      >
        01
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-01-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-02-02-22"
    >
      <p
        className="text-sm p-1 mt-1"
        day-testid="row-value-02-02-22"
      >
        WED
      </p>
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-02-02-22"
      >
        02
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-02-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-03-02-22"
    >
      <p
        className="text-sm p-1 mt-1"
        day-testid="row-value-03-02-22"
      >
        THU
      </p>
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-03-02-22"
      >
        03
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-03-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-04-02-22"
    >
      <p
        className="text-sm p-1 mt-1"
        day-testid="row-value-04-02-22"
      >
        FRI
      </p>
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-04-02-22"
      >
        04
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-04-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-05-02-22"
    >
      <p
        className="text-sm p-1 mt-1"
        day-testid="row-value-05-02-22"
      >
        SAT
      </p>
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-05-02-22"
      >
        05
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-05-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-06-02-22"
    >
      <p
        className="text-sm p-1 mt-1"
        day-testid="row-value-06-02-22"
      >
        SUN
      </p>
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-06-02-22"
      >
        06
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-06-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-07-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-07-02-22"
      >
        07
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-07-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-08-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-08-02-22"
      >
        08
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-08-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-09-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-09-02-22"
      >
        09
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-09-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-10-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-10-02-22"
      >
        10
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-10-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-11-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-11-02-22"
      >
        11
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-11-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-12-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-12-02-22"
      >
        12
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-12-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-13-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-13-02-22"
      >
        13
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-13-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-14-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-14-02-22"
      >
        14
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-14-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-15-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-15-02-22"
      >
        15
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-15-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-16-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-16-02-22"
      >
        16
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-16-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-17-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-17-02-22"
      >
        17
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-17-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-18-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-18-02-22"
      >
        18
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-18-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-19-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-19-02-22"
      >
        19
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-19-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-20-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-20-02-22"
      >
        20
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-20-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-21-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-21-02-22"
      >
        21
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-21-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-22-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-22-02-22"
      >
        22
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-22-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-23-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-23-02-22"
      >
        23
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-23-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-24-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-24-02-22"
      >
        24
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-24-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-25-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-25-02-22"
      >
        25
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-25-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-26-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-26-02-22"
      >
        26
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-26-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-27-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-27-02-22"
      >
        27
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-27-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-28-02-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-28-02-22"
      >
        28
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-28-02-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-01-03-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-01-03-22"
      >
        01
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-01-03-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-02-03-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-02-03-22"
      >
        02
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-02-03-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-03-03-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-03-03-22"
      >
        03
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-03-03-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-04-03-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-04-03-22"
      >
        04
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-04-03-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-05-03-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-05-03-22"
      >
        05
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-05-03-22"
      onClick={[Function]}
    />
  </div>
  <div
    className="border border-gray-200 flex flex-col"
  >
    <header
      className="flex flex-col items-center"
      data-testid="day-header-06-03-22"
    >
      <p
        className="text-sm p-1 my-1 text-center "
        data-testid="day-value-06-03-22"
      >
        06
      </p>
    </header>
    <div
      className="flex-1 cursor-pointer"
      data-testid="select-day-06-03-22"
      onClick={[Function]}
    />
  </div>
</div>
`;
